### DISCUSSION TITLE
How to export schema to a file

### DISCUSSION THREAD

**Discussion by bkstorm** [wp-graphql/wp-graphql#1592] [Q&A] [answered]
Hi everyone.
Is any way to export schema to .graphql or json file?
Here is the URL of graphql requests: http://mywebsite.com/index.php?graphql
I try to run: `gq http://mywebsite.com/index.php?graphql --introspect > schema.graphql` but it throws an exception:

> GraphQL error: GraphQL introspection is not allowed, but the query contained __schema or __type

**Comment by bkstorm** [comment, marked as answer]
Just enable public introspection in Setting :D

**Reply by jasonbahl** [reply]
@bkstorm you can also use [WP CLI](https://wp-cli.org/) with the command `wp graphql generate-static-schema`

**Reply by bkstorm** [reply]
@jasonbahl I'm using graphql code generator to generate react apollo hooks, but it throws `Syntax Error: Unexpected Name "TermNode"` error, I check the schema.graphql, everything seems to be OK. Do you know why?
[https://imgur.com/FpnBYPg](https://imgur.com/FpnBYPg)
